core/services/keystore/eth.go

```diff
@@ -244,22 +248,32 @@ RETURNING id, next_nonce, address, evm_chain_id, disabled, created_at, updated_a
 	return nil
 }
 
-func (ks *eth) Disable(address common.Address, chainID *big.Int) error {
-	_, err := ks.orm.q.Exec(`UPDATE evm_key_states SET disabled = true WHERE address = $1 AND evm_chain_id = $2`, address, chainID.String())
+func (ks *eth) Disable(address common.Address, chainID *big.Int, qopts ...pg.QOpt) error {
+	ks.lock.Lock()
+	defer ks.lock.Unlock()
+	_, found := ks.keyRing.Eth[address.Hex()]
+	if !found {
+		return errors.Errorf("no key exists with ID %s", address.Hex())
+	}
+	return ks.disable(address, chainID)
```

Comment:
Let's pass the `qopts` value here.

Reply:
Hmm, I'm getting deja vu here... didn't I fix this already?

Reply:
Ah, forgot to pull in the latest commit. Check again pls @vyzaldysanchez